google/resource_cloudiot_registry_test.go

```diff
@@ -0,0 +1,206 @@
+	t.Parallel()
+
+	registryName := fmt.Sprintf("psregistry-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudiotRegistryDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCloudiotRegistry_basic(registryName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCloudiotRegistryExists(
+						"google_cloudiot_registry.foobar"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudiotRegistryCreate_extended(t *testing.T) {
+	t.Parallel()
+
+	registryName := fmt.Sprintf("psregistry-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudiotRegistryDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCloudiotRegistry_extended(registryName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCloudiotRegistryExists(
+						"google_cloudiot_registry.foobar"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudiotRegistryUpdate(t *testing.T) {
+	t.Parallel()
+
+	registryName := fmt.Sprintf("psregistry-test-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudiotRegistryDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCloudiotRegistry_basic(registryName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCloudiotRegistryExists(
+						"google_cloudiot_registry.foobar"),
+				),
+			},
+			resource.TestStep{
+				Config: testAccCloudiotRegistry_extended(registryName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCloudiotRegistryUpdate(
+						"google_cloudiot_registry.foobar"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckCloudiotRegistryDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_cloudiot_registry" {
+			continue
+		}
+
+		config := testAccProvider.Meta().(*Config)
+		registry, _ := config.clientCloudiot.Projects.Locations.Registries.Get(rs.Primary.ID).Do()
+		if registry != nil {
+			return fmt.Errorf("Registry still present")
+		}
+	}
+
+	return nil
+}
+
+func testAccCloudiotRegistryExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+		config := testAccProvider.Meta().(*Config)
+		_, err := config.clientCloudiot.Projects.Locations.Registries.Get(rs.Primary.ID).Do()
+		if err != nil {
+			return fmt.Errorf("Registry does not exist")
+		}
+
+		return nil
+	}
+}
+
+func testAccCloudiotRegistryUpdate(n string) resource.TestCheckFunc {
```

Comment:
This seems elaborate - I think it only tests that the state matches the config, which is implicitly part of a test that has config.  If the state doesn't match the config post-update, the test will fail even without this method, and so removing this method reduces the difficulty of maintenance.

Reply:
👍 Have removed this function. I was not aware of this internal test.

Reply:
Oh, I apologize but I was wrong about this - this is true only when there's an importstate step on the test.  If you can add one of those in, these tests will be easier to maintain, but if that's a hassle for unforeseen reasons you can definitely go back to this method.  My bad!